apps/builder/app/builder/shared/sync/sync-server.ts

```diff
@@ -62,27 +65,53 @@ const useNewEntriesCheck = ({
         return;
       }
 
-      enqueue(() =>
-        fetch(restPatchPath({ authToken }), {
+      enqueue(async () => {
+        // save the initial version from loaded build
+        if (lastVersion === undefined) {
+          lastVersion = version;
+        }
+        const response = await fetch(restPatchPath({ authToken }), {
           method: "post",
           body: JSON.stringify({
             transactions,
             buildId,
             projectId,
+            // provide latest stored version to server
+            version: lastVersion,
           }),
-        })
-      );
+        });
+        if (response.ok) {
+          const result = await response.json();
+          if (result.status === "ok") {
+            lastVersion += 1;
+            return { ok: true };
+          }
+          // when versions mismatched ask user to reload
+          // user may cancel to copy own state before reloading
+          if (result.status === "version_mismatched") {
+            const shouldReload = confirm(
+              "Someone else is edited the project. Please reload to get the latest state."
```

Comment:
"Project was edited in a different tab, browser or by another user. Please reload to get the latest state."

Notion's AI works great for stuff like this, I asked it to improve the text


<img width="876" alt="Screenshot 2023-05-29 at 10 19 00" src="https://github.com/webstudio-is/webstudio-builder/assets/52824/714c2330-5584-47c8-8c10-73252dfcd291">